pkg/cloud/config.go

```diff
@@ -0,0 +1,188 @@
+package cloud
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+)
+
+type ConnectionConfig struct {
+	// Kind is a string value representing the REST resource this object represents.
+	// Servers may infer this from the endpoint the client submits requests to.
+	// In CamelCase.
+	// +optional
+	Kind string `yaml:"kind,omitempty"`
+	// APIVersion defines the versioned schema of this representation of an object.
+	// Servers should convert recognized schemas to the latest internal value, and
+	// may reject unrecognized values.
+	// +optional
+	APIVersion string `yaml:"apiVersion,omitempty"`
+	// Datacenters is a map of referencable names to datacenter configs.
+	Datacenters map[string]*Datacenter `yaml:"datacenters"`
+	// AuthInfos is a map of referencable names to authentication configs.
+	AuthInfos map[string]*AuthInfo `yaml:"authInfos"`
+	// Contexts is a map of referencable names to context configs.
+	Contexts map[string]*Context `yaml:"contexts"`
+	// CurrentContext is the name of the context that you would like to use by default.
+	CurrentContext string `yaml:"currentContext"`
```

Comment:
What is the purpose of dynamically changing the context?
It is not even a use case if you thy changing it there is a race.

Reply:
Gotcha.
See my suggestion on making this package capable of producing gocql.Cluster and being just a helper.